### PR TITLE
Delete edit note

### PR DESCRIPTION
**Pull request by jamarob**
add delete and edit note feature

### CODE REVIEW

File: src/pages/DeleteNotePage.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+export default function DeleteNotePage({ findNote, deleteNote }) {
+  const history = useHistory()
+  const { id } = useParams()
+  const { text } = findNote(id)
+  return (
+    <>
+      <Header title="Delete Note" />
+      <Main>
+        <h2>Are you shure you want to delete this note?</h2>
```
Comment (Antuofermo) [inline src/pages/DeleteNotePage.js]: shure ist ein Brand, sure ist ein Adjektiv ;)

File: src/pages/DeleteNotePage.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        <p>{text}</p>
+        <ConfirmationButtons>
+          <button onClick={handleCancel} className="cancel">
```
Comment (Antuofermo) [inline src/pages/DeleteNotePage.js]: Add cursor: pointer to the style

File: src/pages/DeleteNotePage.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            No
+          </button>
+          <button onClick={handleDelete} className="confirm">
```
Comment (Antuofermo) [inline src/pages/DeleteNotePage.js]: Same as cancel btn

File: src/pages/DeleteNotePage.js (hunk 1 above)
Comment (jamarob) [reply src/pages/DeleteNotePage.js]: Yeah, sure.. :D

File: src/pages/DeleteNotePage.js (hunk 2 above)
Comment (jamarob) [reply src/pages/DeleteNotePage.js]: Good idea, thanks!

File: src/components/Note.js [hunk 4]
```diff
@@ -3,17 +3,31 @@ import styled from 'styled-components/macro'
...
 }
 
-export default function Note({ created, text, onTagClick }) {
+export default function Note({ id, created, edited, text, onTagClick }) {
   return (
     <StyledNote>
-      <DateView time={created} />
+      <NoteHeader>
+        <DateView created={created} edited={edited} />
+        <ActionButtons>
```
[the remark this answers is not in the document]
Comment (jamarob) [reply src/components/Note.js]: You are right, I will rename them.

File: src/components/DateView.js [hunk 5]
```diff
@@ -3,11 +3,16 @@ import styled from 'styled-components/macro'
...
+  created: PropTypes.number.isRequired,
+  edited: PropTypes.number,
 }
 
-export default function DateView({ time }) {
-  return <StyledDate>{toLocaleString(time)}</StyledDate>
+export default function DateView({ created, edited }) {
+  return (
+    <StyledDate>
+      {toLocaleString(created)} {edited && ' · ' + toLocaleString(edited)}
```
[the remark this answers is not in the document]
Comment (jamarob) [reply src/components/DateView.js]: Good idea!

File: src/components/NotesList.js [hunk 6]
```diff
@@ -25,7 +25,9 @@ export default function NotesList({ notes, onTagClick }) {
     return notes.map(note => (
       <Note
         key={note.id}
+        id={note.id}
```
[the remark this answers is not in the document]
Comment (jamarob) [reply src/components/NotesList.js]: `note.id` is a unique string, so I use it for the key prop and pass it down to be used later in a callback. As far as I know key is not really a prob and can not be accessed inside the component, hence the duplication.

**Comment by jamarob** [comment]
Thanks Mila!

**Pull request merged**